src/engine/srv.c

```diff
@@ -800,17 +800,20 @@ dss_xstreams_fini(bool force)
 	bool			 started = false;
 
 	D_DEBUG(DB_TRACE, "Stopping execution streams\n");
-	dss_xstreams_open_barrier();
-	rc = bio_nvme_ctl(BIO_CTL_NOTIFY_STARTED, &started);
-	D_ASSERT(rc == 0);
 
+	ABT_mutex_lock(xstream_data.xd_mutex);
 	/* Notify all xstreams to reject new ULT creation first */
 	for (i = 0; i < xstream_data.xd_xs_nr; i++) {
 		dx = xstream_data.xd_xs_ptrs[i];
 		if (dx == NULL)
 			continue;
 		ABT_future_set(dx->dx_stopping, dx);
 	}
+	ABT_cond_broadcast(xstream_data.xd_ult_barrier);
+	ABT_mutex_unlock(xstream_data.xd_mutex);
+
+	rc = bio_nvme_ctl(BIO_CTL_NOTIFY_STARTED, &started);
```

Comment:
and not sure what's this call's purpose?

BTW, may need to confirm with the person who reported the ticket if the problem is resolved. Thx.

Reply:
@NiuYawei Could you please shed some light about the Xuezhao's question?
I will check with the user to see whether the problem is resolved or not. Thank you!

Reply:
I don't think there is any race in current implementation, please check my comment on Jira ticket.

Reply:
@NiuYawei ok. I will check it out. Thank you very much!